gpu/octree/src/cuda/radius_search.cu

```diff
@@ -331,14 +373,16 @@ namespace pcl
 }
 
 template<typename BatchType>
-void pcl::device::OctreeImpl::radiusSearchEx(BatchType& batch, const Queries& queries, NeighborIndices& results)
+void pcl::device::OctreeImpl::radiusSearchEx(BatchType& batch, const Queries& queries, NeighborIndices& results, BatchResultSqrDists& sqr_distances, const bool find_dist)
 {
     batch.indices = indices;
     batch.octree = octreeGlobal;
 
+    batch.find_dist = find_dist;
     batch.max_results = results.max_elems;
     batch.output = results.data;                
     batch.output_sizes = results.sizes;
+    batch.sqr_distances = sqr_distances;
 
     batch.points = points_sorted;
```

Comment:
Notice that `batch` should have been populated with all required variables before invoking the function

```cpp
    batch.indices = indices;
    batch.octree = octreeGlobal;

    batch.find_dist = find_dist;
    batch.max_results = results.max_elems;
    batch.output = results.data;                
    batch.output_sizes = results.sizes;
    batch.sqr_distances = sqr_distances; 

    //and only then
   radiusSearchEx(batch); // no need for extra arguments because everything is inside batch
```

Reply:
Wouldn't that require duplicating that code across all 3 of these radius search methods?
```cpp
void pcl::device::OctreeImpl::radiusSearch(const Queries& queries, const float radius, NeighborIndices& results, BatchResultSqrDists& sqr_distances, const bool compute_all_distances)
{        
    using BatchType = Batch<SharedRadius, DirectQuery>;

    BatchType batch;
    batch.radius = radius;
    batch.queries = queries;
    radiusSearchEx(batch, queries, results, sqr_distances, compute_all_distances);
}

void pcl::device::OctreeImpl::radiusSearch(const Queries& queries, const Radiuses& radiuses, NeighborIndices& results, BatchResultSqrDists& sqr_distances, const bool compute_all_distances)
{
    using BatchType = Batch<IndividualRadius, DirectQuery>;

    BatchType batch;
    batch.radiuses = radiuses;
    batch.queries = queries;
    radiusSearchEx(batch, queries, results, sqr_distances, compute_all_distances);
}

void pcl::device::OctreeImpl::radiusSearch(const Queries& queries, const Indices& indices, const float radius, NeighborIndices& results, BatchResultSqrDists& sqr_distances, const bool compute_all_distances)
{
    using BatchType = Batch<SharedRadius, IndicesQuery>;

    BatchType batch;
    batch.radius = radius;
    
    batch.queries = queries;
    batch.queries_indices = indices;
    batch.queries.size = indices.size();

    radiusSearchEx(batch, queries, results, sqr_distances, compute_all_distances);
}
```
I think the original intention was to move all the common operations into `radiusSearchEx()` and only use those 3 methods to handle the differences between them in terms of queries, indices and radii.